mautic/mautic
# mt(...) tracking function can be called multiple times per page load to support SPA

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Y
| New feature? | N
| Related user documentation PR URL | /
| Related developer documentation PR URL | /
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/1861
| BC breaks? | N
| Deprecations? | N

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Single Page Applications (SPA) need to call the `mt('send', 'pageview');` JS function multiple times per page load, because the pages are loaded via JS. Right now it could be executed only once per page load. This PR allows to call it multiple times per page load.


#### Testing page:
```
<html>
    <head>
        <title>Test of the Mautic JS tracking</title>
        <script>
            console.log('JS executed');
            
            (function(w,d,t,u,n,a,m){w['MauticTrackingObject']=n;
                w[n]=w[n]||function(){(w[n].q=w[n].q||[]).push(arguments)},a=d.createElement(t),
                m=d.getElementsByTagName(t)[0];a.async=1;a.src=u;m.parentNode.insertBefore(a,m)
            })(window,document,'script','http://[fill in your mautic URL here]/mtc.js','mt');

            mt(
                'send',
                'pageview', {email: 'JStrackingtest@gmail.com'}
            );

            window.setTimeout(function() {
                console.log('send the second pageview');
                mt('send', 'pageview');
            }, 4000);
        </script>
    </head>
    <body>
        JS test. Open the JS console to see the results.
    </body>
</html>
```
#### Steps to test this PR:
1. Apply this PR.
2. Test again.
3. Notice that the tracking pixel is loaded right after page load and again after 4 seconds. An email is loaded in the first request, not in the second.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Create a HTML page and fill in the testing content above.
2. The script will try load the tracking pixel on page load and then again after 4 seconds.
3. Notice in the dev tools that the tracking pixel is called only once.
